src/extensions/registries/page-registry.ts

```diff
@@ -44,7 +44,7 @@ export function getExtensionPageUrl<P extends object>({ extensionId, pageId = ""
   const extensionBaseUrl = compile(`/extension/:name`)({
     name: sanitizeExtensionName(extensionId), // compile only with extension-id first and define base path
   });
-  const extPageRoutePath = path.join(extensionBaseUrl, pageId); // page-id might contain route :param-s, so don't compile yet
+  const extPageRoutePath = path.join(extensionBaseUrl, pageId).replace(/\\/g, "/"); // path.join returns \ as separator on Windows and needs to be replaced with /
```

Comment:
It doesn't seem correct to use path.join (intended for file paths) for joining the url. See https://stackoverflow.com/a/16301947

Reply:
it seems that we cannot achieve the same with `new URL()` since we are combining the path part of the url. However `path.posix.join` might be something that we want to use.

Reply:
path.posix.join also doesn't sound like optimal here, even if `new URL()` is insufficient. Surely there's some npm dependency to join url parts properly.